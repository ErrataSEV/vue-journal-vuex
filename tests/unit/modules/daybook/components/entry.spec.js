import { shallowMount } from "@vue/test-utils";

import Entry from '@/modules/daybook/components/EntryComponent';
import { journalState } from "../../../mock-data/test-journal-state";

describe('Test on Entry Component', () => {

    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount(Entry, {
        global:{
            mocks: {
                $router: mockRouter
            }
        },
        props: {
            entry: journalState.entries[1]
        }
    })

    test('must be equal to the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('must redirect on click to the entry-container', () => {

        const entryConatiner = wrapper.find('.entry-container')
        entryConatiner.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith({
            name: "entry",
            params: {
                id: journalState.entries[1].id
            }
        })
    })

    test('test on properties computed', () => {

        expect( wrapper.vm.day ).toBe(26)
        expect( wrapper.vm.month ).toBe('Marzo')
        expect( wrapper.vm.yearDay ).toBe('2023, Domingo')
    })
})