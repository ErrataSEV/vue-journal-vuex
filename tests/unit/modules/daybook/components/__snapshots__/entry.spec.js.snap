// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Entry Component must be equal to the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">26</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-light">2023, Domingo</span></div>
  <div class="entry-description">I'm learning vue with Fernando Herrera</div>
</div>
`;
